component/src/main/java/org/wso2/extension/siddhi/io/file/FileSource.java

```diff
@@ -0,0 +1,539 @@
+        } else {
+            actionAfterProcess = optionHolder.validateAndGetStaticValue(Constants.ACTION_AFTER_PROCESS,
+                    Constants.DELETE);
+        }
+        actionAfterFailure = optionHolder.validateAndGetStaticValue(Constants.ACTION_AFTER_FAILURE, Constants.DELETE);
+        if (optionHolder.isOptionExists(Constants.MOVE_AFTER_PROCESS)) {
+            moveAfterProcess = optionHolder.validateAndGetStaticValue(Constants.MOVE_AFTER_PROCESS);
+        }
+        if (optionHolder.isOptionExists(Constants.MOVE_AFTER_FAILURE)) {
+            moveAfterFailure = optionHolder.validateAndGetStaticValue(Constants.MOVE_AFTER_FAILURE);
+        }
+
+        dirPollingInterval = optionHolder.validateAndGetStaticValue(Constants.DIRECTORY_POLLING_INTERVAL, "1000");
+
+        filePollingInterval = optionHolder.validateAndGetStaticValue(Constants.FILE_POLLING_INTERVAL, "1000");
+
+        beginRegex = optionHolder.validateAndGetStaticValue(Constants.BEGIN_REGEX, null);
+        endRegex = optionHolder.validateAndGetStaticValue(Constants.END_REGEX, null);
+
+        validateParameters();
+        createInitialSourceConf();
+        updateSourceConf();
+    }
+
+
+    @Override
+    public Class[] getOutputEventClasses() {
+        return new Class[]{String.class, byte[].class};
+    }
+
+    @Override
+    public void connect(ConnectionCallback connectionCallback) throws ConnectionUnavailableException {
+        updateSourceConf();
+        deployServers();
+    }
+
+    @Override
+    public void disconnect() {
+        try {
+            if (fileSystemServerConnector != null) {
+                fileSystemServerConnector.stop();
+                fileSystemServerConnector = null;
+            }
+            //fileSystemServerConnector = null;
+            if (isTailingEnabled) {
+                fileSourceConfiguration.getFileServerConnector().stop();
+                fileSourceConfiguration.setFileServerConnector(null);
+            }
+        } catch (ServerConnectorException e) {
+           throw new SiddhiAppRuntimeException("Failed to stop the file server : " + e.getMessage());
+        }
+    }
+
+
+    public void destroy() {
+
+    }
+
+    public void pause() {
+        try {
+            if (fileSystemServerConnector != null) {
+                fileSystemServerConnector.stop();
+            }
+            if (isTailingEnabled && fileSourceConfiguration.getFileServerConnector() != null) {
+                fileSourceConfiguration.getFileServerConnector().stop();
+            }
+        } catch (ServerConnectorException e) {
+            throw new SiddhiAppRuntimeException("Failed to stop the file server : " + e.getMessage());
+        }
+    }
+
+    public void resume() {
+        try {
+            updateSourceConf();
+            deployServers();
+        } catch (ConnectionUnavailableException e) {
+            throw new SiddhiAppRuntimeException("Failed to resume siddhi app runtime due to " + e.getMessage(), e);
```

Comment:
Better mention siddhi app name in the log.

Reply:
Done by 0b41464d4ebc6683c78635fe198ceba602f845e6